examples/function.jl

```diff
@@ -0,0 +1,27 @@
+using KernelDensity, AbstractPlotting, Makie
+import AbstractPlotting: convert_arguments, PlotFunc, plottype
+to_tuple(t::Tuple) = t
+to_tuple(t) = (t,)
+
+
+function convert_arguments(P::PlotFunc, f::Function, args...; kwargs...)
+    tmp = f(args...; Iterators.filter(t -> last(t) != automatic, kwargs)...) |> to_tuple
+    convert_arguments(P, tmp...)
+end
+
+# remove convert_arguments(P, f, x) = (x, f.(x))
+function convert_arguments(P::PlotFunc, d::KernelDensity.UnivariateKDE)
+    ptype = plottype(P, Lines) # choose the more concrete one
+    ptype => convert_arguments(ptype, d.x, d.density)
+end
+
+function convert_arguments(P::Type{<: Combined{T}}, d::KernelDensity.BivariateKDE) where T
+    ptype = plottype(P, Heatmap)
+    ptype => convert_arguments(ptype, d.x, d.y, d.density)
```

Comment:
Is `convert_arguments` always expected to return a pair at the end? Or is there a `to_pair` function that would turn it into a pair with the first element the "undefined plot" (I guess `Combined{Any, <:Any}`)?

Reply:
No, the pair is optional, but you want to define a default plot type for BivariateKDE, so you need to return a pair with the plot type;)